x-pack/elastic-agent/_meta/config/common.p2.yml.tmpl

```diff
@@ -33,6 +33,11 @@ inputs:
 #   logs: true
 #   # enables metrics monitoring
 #   metrics: true
+#   # exposes stats on specified HTTP port
+#   # process stats are exposed only using this option
+#   # it is up to a caller to make sure port is usable and free to use.
+#   # by default 0 is used meaning socket is used instead.
+#   port: 0
```

Comment:
I suggest we use the same config blocks as we have in beats: https://github.com/elastic/beats/blob/master/filebeat/filebeat.reference.yml#L2507

We need a `host` to decide if it should only be exposed on localhost or broader. This also allows to add the `enabled` option.

Reply:
for reference this is being discussed here: https://github.com/elastic/cloud/issues/74800#issuecomment-809293852

Reply:
Right, at least for when it is run in docker `localhost` would not be sufficient.

Reply:
updated with http.enabled/host/port options

Reply:
Great. @simitt I assume on Cloud we can just use this in the template and make the port configurable or hardcode.